contracts/MultiMerkleMine.sol

```diff
@@ -0,0 +1,62 @@
+pragma solidity ^0.4.24;
+
+import "./MerkleMine.sol";
+import "zeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+
+contract MultiMerkleMine {
+
+	using SafeMath for uint256;
+
+	bytes[] private _proofs;
+
+	event AlreadyGenerated(address indexed recipient, address indexed caller);
+
+	function extractProofs(bytes _merkleProofs) internal returns(bytes[]) {
+		require(_merkleProofs.length != 0, 'No proofs supplied!');
+		for(uint256 i=0; i<_merkleProofs.length; i++){
+		    uint256 proofSize = uint256(_merkleProofs[i]);
+		    require(proofSize % 32 == 0, 'Invalid proof detected!');
+		    bytes memory _proof = new bytes(proofSize);
+		    uint256 j = 0;
+		    while(j<proofSize){
+		        _proof[j]=_merkleProofs[i+1];
+		        i+=1;
+		        j+=1;
```

Comment:
Is it best practice to use SafeMath even for incrementing counters in loops, or is that overkill?

Reply:
Using SafeMath is always a tradeoff b/w higher gas costs and security. IMO, with loop counter variables it would be overkill as its usually easy to verify their bounds.